File: cpp/math/Euclidean.cpp
#include <iostream>
using namespace std;

class Euclidean {
public:
    int gcd(int num1, int num2) {
        return !num2 ? num1 : gcd(num2, num1 % num2);
    }

    int lcm(int num1, int num2) {
        return num1 / gcd(num1, num2) * num2;
    }
};

File: py/math/Euclidean.py
#!/usr/bin/env python
# -*- coding:gbk -*-

class Euclidean:
    def gcd(self, num1: int, num2: int) -> int:
        return num1 if not num2 else self.gcd(num2, num1 % num2)

    def lcm(self, num1: int, num2: int) -> int:
        return num1 // self.gcd(num1, num2) * num2
